docs/guides/javascript/reactive/index.md

```diff
@@ -0,0 +1,1382 @@
+
+Components can do many things but, in general, they are not hard to implement because they are always structured in the same way and the BaseComponent class does all the hard part for us. There are only a few scenarios where a reactive component can be more complicated than a traditional JS undefined pattern code and, even in those cases, the maintainability is much better.
+
+Let's take a look at the component anatomy.
+
+### Extending the BaseComponent class
+
+All reactive UIs are object-oriented and all components extend the BaseComponent class. This means that the default component structure is:
+
+<details>
+  <summary>View example</summary>
+  <div>
+
+```js title="path/to/plugin/amd/src/yourcomponent.js"
+import {BaseComponent} from 'core/reactive';
+
+export default class extends BaseComponent {
+(...)
+}
+```
+
+  </div>
+</details>
+
+### Instantiate a Component
+
+Any component belongs to one DOM element and is registered in a Reactive Instance. This means that to create a new instance we need to provide at least those two elements. For example:
+
+<details>
+  <summary>View example</summary>
+  <div>
+
+```js title="path/to/plugin/amd/src/main.js"
+import {reactiveInstance} from 'YOUR_PLUGIN/reactive';
+import {YourComponent} from 'YOUR_PLUGIN/yourcomponent;
+
+(...)
+
+return new YourComponent({
+     element: document.getElementById(DOM_ELEMENT_ID),
+     reactive: reactiveInstance,
+ (... other data your component needs ...)
+});
+```
+
+  </div>
+</details>
+
+In case no Reactive Instance is passed, the BaseComponent will try to inherit the instance from a parent DOM element. This way subcomponents can omit that param.
+
+### Initialize a component from a mustache template
+
+Components are easy to embed in mustache files. To do so your class must have a static "init" method that could be called inside the {{#JS}}.
+
+<Tabs>
+  <TabItem value="component-view" label="Component class" default>
+
+```js title="path/to/plugin/amd/src/childcomponent.js"
+import {BaseComponent} from 'core/reactive';
+import {reactiveInstance} from 'YOUR_PLUGIN/reactive';
+
+export default class extends BaseComponent {
+    static init(target, selectors) {
+        return new Component({
+            element: document.getElementById(target),
+            reactive: reactiveInstance,
+            selectors,
+        });
+    }
+    // (...)
+}
+```
+
+  </TabItem>
+  <TabItem value="mustache-view" label="Mustache file">
+
+```handlebars title="path/to/plugin/templates/yourthing/childcomponent.mustache"
+<div id="{{uniqid}}-your-plugin-{{id}}"> (...) </div>
+{{#js}}
+require(['YOUR_PLUGIN/local/yourthing/childcomponent'], function(component) {
+    component.init('{{uniqid}}-your-plugin-{{id}}');
+});
+{{/js}}
+```
+
```

Comment:
It might be worth adding a tip admonitino to expalin the use of uniqid here.

```suggestion

:::tip Generating unique id attributes

You can use the `{{uniqid}}` Mustache helper within your code to help you generate a unique id for your HTML attributes and target them in your React component.

Please note that the `{{uniqid}}` helper generates a single value each time it is rendered, and you will need to combine it with other data to create a truly unique value. In this example the name of the plugin, and a static element `id` is used to generate a unique value.

:::

```